src/Control/SortControl.php

```diff
@@ -169,12 +169,13 @@ public function getSort()
      * Sort the given query according to the request
      *
      * @param Query $query
+     * @param ?array|string $defaultSort
      *
      * @return $this
      */
-    public function apply(Query $query)
+    public function apply(Query $query, $defaultSort = null)
     {
-        $default = (array) $query->getModel()->getDefaultSort();
+        $default = $defaultSort ?? (array) $query->getModel()->getDefaultSort();
```

Comment:
Previously `$default` was always an array, even if empty, due to the cast. Now if `$defaultSort` is not an array, `$default` isn't as well. I'm not sure the other code can handle this, have you checked?

Reply:
There is no change needed as `SortUtil::normalizeSortSpec` accepts both array or string as parameters and it is the only place where `$default` is used.